src/main/java/org/candlepin/subscriptions/tally/InventorySwatchDataCollator.java

```diff
@@ -91,10 +91,15 @@ public InventorySwatchDataCollator(
    */
   public int collateData(String orgId, int culledOffsetDays, Processor processor) {
     Stream<InventoryHostFacts> inventorySystemStream =
-        inventoryRepository.streamFacts(orgId, culledOffsetDays);
+        inventoryRepository
+            .streamFacts(orgId, culledOffsetDays)
+            .sorted(Comparator.comparing(SortKey::fromHbiSystem));
```

Comment:
This should be a no-op given the order by clause:
https://github.com/RedHatInsights/rhsm-subscriptions/blob/9050d6340388cc41ee15c3f771cd96ab15519555/src/main/java/org/candlepin/subscriptions/inventory/db/model/InventoryHost.java#L159

should be equivalent to the `SortKey`'s `compareTo` implementation:

https://github.com/RedHatInsights/rhsm-subscriptions/blob/9050d6340388cc41ee15c3f771cd96ab15519555/src/main/java/org/candlepin/subscriptions/tally/InventorySwatchDataCollator.java#L268-L281